HYDRA-691 : Move mayaUsd code for managing stages inside mayaHydra (#38)
Autodesk/maya-hydra: lib/mayaHydra/hydraExtensions/CMakeLists.txt

# -----------------------------------------------------------------------------
# compiler configuration
# -----------------------------------------------------------------------------

> if(DEFINED MAYAUSD_LOCATION)

Reviewer: I don't think you need this conditional, could simply be
```suggestion
$<$<BOOL:${MAYAUSD_LOCATION}>:MAYAHYDRALIB_MAYAUSD_ENABLED>
```

    target_compile_definitions(${TARGET_NAME} 
        PRIVATE 
            MAYAHYDRALIB_EXPORT

>             MAYAHYDRALIB_MAYAUSD_IS_USED_TO_BUILD

Reviewer: A bit wordy for a boolean variable.  How about:
```suggestion
            MAYAHYDRALIB_MAYAUSD_ENABLED
```

> if(DEFINED MAYAUSD_LOCATION)
>     target_compile_definitions(${TARGET_NAME} 
>         PRIVATE 
>             MAYAHYDRALIB_EXPORT
>             MAYAHYDRALIB_MAYAUSD_IS_USED_TO_BUILD
>             $<$<BOOL:${IS_MACOSX}>:OSMac_>
>             # Copy-pasted from lib/mayaUsd/CMakeLists.txt
>     )
> else()
>     target_compile_definitions(${TARGET_NAME} 
>         PRIVATE 
>             MAYAHYDRALIB_EXPORT
>             $<$<BOOL:${IS_MACOSX}>:OSMac_>
>             # Copy-pasted from lib/mayaUsd/CMakeLists.txt
>     )
> endif()

Reviewer 2: Once you create the FindMayaUsd.cmake it will set MAYAUSD_FOUND (to TRUE for cmake). Then this can be simplified with a cmake generator expression.
```suggestion
target_compile_definitions(${TARGET_NAME} 
    PRIVATE 
        MAYAHYDRALIB_EXPORT
        $<$<BOOL:${IS_MACOSX}>:OSMac_>
        $<$<BOOL:${MAYAUSD_FOUND}>:MAYAHYDRALIB_MAYAUSD_IS_USED_TO_BUILD>
        # Copy-pasted from lib/mayaUsd/CMakeLists.txt
)
```

        
mayaHydra_compile_config(${TARGET_NAME})

# -----------------------------------------------------------------------------
# include directories
# -----------------------------------------------------------------------------

> if(DEFINED MAYAUSD_LOCATION)

Reviewer: See above using a CMake generator expression.

> if(DEFINED MAYAUSD_LOCATION)
> target_include_directories(${TARGET_NAME} 
>             PUBLIC
>                 ${MAYA_INCLUDE_DIRS}
>                 ${PYTHON_INCLUDE_DIR}
>                 ${PXR_INCLUDE_DIRS}
>                 ${BOOST_INCLUDE_DIR}
>                 ${CMAKE_BINARY_DIR}/include
>         ${MAYAUSD_LOCATION}/include
>             PRIVATE
>                 $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
> )
>      else()
>         target_include_directories(${TARGET_NAME} 
>             PUBLIC
>                 ${MAYA_INCLUDE_DIRS}
>                 ${PYTHON_INCLUDE_DIR}
>                 ${PXR_INCLUDE_DIRS}
>                 ${BOOST_INCLUDE_DIR}
>                 ${CMAKE_BINARY_DIR}/include
>             PRIVATE
>                 $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
>         )
>      endif()

Reviewer 2: Can simplify with generator expression.
```suggestion
    target_include_directories(${TARGET_NAME} 
        PUBLIC
            ${MAYA_INCLUDE_DIRS}
            ${PYTHON_INCLUDE_DIR}
            ${PXR_INCLUDE_DIRS}
            ${BOOST_INCLUDE_DIR}
            ${CMAKE_BINARY_DIR}/include
        PRIVATE
            $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>
            $<$<BOOL:${MAYAUSD_FOUND}>:${MAYAUSD_INCLUDE_DIR}>
    )
```

# -----------------------------------------------------------------------------
# compiler configuration
# -----------------------------------------------------------------------------
target_compile_definitions(${TARGET_NAME} 
    PRIVATE 
        MAYAHYDRALIB_EXPORT

>         $<$<BOOL:${MAYAUSDAPI_LIBRARY}>:MAYAHYDRALIB_MAYAUSDAPI_ENABLED>

Reviewer 2: ```suggestion
        $<$<BOOL:${MayaUsd_FOUND}>:MAYAHYDRALIB_MAYAUSDAPI_ENABLED>
```

        $<$<BOOL:${IS_MACOSX}>:OSMac_>
        # Copy-pasted from lib/mayaUsd/CMakeLists.txt
)
        
mayaHydra_compile_config(${TARGET_NAME})

# -----------------------------------------------------------------------------
# include directories
# -----------------------------------------------------------------------------
target_include_directories(${TARGET_NAME} 
            PUBLIC
                ${MAYA_INCLUDE_DIRS}
                ${PYTHON_INCLUDE_DIR}
                ${PXR_INCLUDE_DIRS}
                ${BOOST_INCLUDE_DIR}
                ${CMAKE_BINARY_DIR}/include
            PRIVATE
                $<$<BOOL:${UFE_FOUND}>:${UFE_INCLUDE_DIR}>

>                 $<$<BOOL:${MAYAUSDAPI_LIBRARY}>:${MAYAUSD_INCLUDE_DIR}>

Reviewer 2: ```suggestion
                $<$<BOOL:${MayaUsd_FOUND}>:${MAYAUSD_INCLUDE_DIR}>
```

    )
endif()

# -----------------------------------------------------------------------------
# link libraries
# -----------------------------------------------------------------------------
target_link_libraries(${TARGET_NAME} 
    PUBLIC
        ar
        gf
        hd
        hdx
        js
        kind
        plug
        sdf
        tf
        trace
        usd
        usdGeom
        usdImaging
        usdImagingGL
        usdLux
        usdRender
        usdRi
        usdShade
        usdSkel
        usdUtils
        ufeExtensions
        flowViewport
        ${MAYA_LIBRARIES}
    PRIVATE
        $<$<BOOL:${USD_001905_BUILD}>:hio>
        ${PYTHON_LIBRARIES} 
        $<$<BOOL:${UFE_FOUND}>:${UFE_LIBRARY}>

>         $<$<BOOL:${MAYAUSDAPI_LIBRARY}>:${MAYAUSDAPI_LIBRARY}>

Reviewer 2: ```suggestion
        $<$<BOOL:${MayaUsd_FOUND}>:${MAYAUSDAPI_LIBRARY}>
```

    # -----------------------------------------------------------------------------
    # link libraries
    # -----------------------------------------------------------------------------

> if(DEFINED MAYAUSD_LOCATION)

Reviewer: Again, a generator expression should be used.

    find_library(MAYAUSD_API_LIBRARY NAMES mayaUsdAPI PATHS ${MAYAUSD_LOCATION}/lib NO_DEFAULT_PATH)
    if(NOT MAYAUSD_API_LIBRARY)
        message(WARNING "Could not find mayaUsdAPI library.")

>     endif()

Reviewer 2: This should be in the new FindMayaUsd.cmake

Reviewer 2: Can use generator expression again to add the library (if found). No need for the `if(DEFINED MAYAUSD_LOCATION)`

            $<$<BOOL:${MAYAUSD_FOUND}>:${MAYAUSD_API_LIBRARY}>

Reviewer 2: I'm pretty sure the `if(DEFINED MAYAUSD_VERSION)` can be simplified down to a single generator expression line as well. This might/should work:
`$<$<BOOL:${MAYAUSD_FOUND}>:MAYAUSD_VERSION=${MAYAUSD_VERSION}>`
